### ISSUE TITLE
Accessibility - StreamField renders an empty heading (h3) element

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#9180]
### Issue Summary

* `StreamField` items render a block container with a `h3` element but this title will always be empty, providing no information to the user for the titles in this area of the UI.
* This blocks a critical AA accessibility criteria 2.4.6 Headings & Labels - https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels

### Steps to Reproduce

1. Open the bakerydemo
2. Open any page with a StreamField being used
3. Use an accessibility validator tool or use a screen reader to navigate to one inner stream field block
4. Expected: h3 should read out a title
5. Actual: h3 is empty

- I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (no)
- Found while investigating a solution for #8673 

### Screenshots

**See Firefox accessibility tool - the header under 'page body' is blank**

<img width="1312" alt="Screen Shot 2022-09-11 at 2 52 22 pm" src="https://user-images.githubusercontent.com/1396140/189513026-fb4514dc-48cc-4feb-aeb2-82013d280f57.png">

**See NVDA - it reads the outer section's title 'Page body' but not the inner one 'paragraph block'**

<img width="1673" alt="Screen Shot 2022-09-11 at 2 50 55 pm" src="https://user-images.githubusercontent.com/1396140/189513036-9329834c-99aa-4e92-95b1-000fe66789bd.png">

**See VoiceOver - does not read any heading for the paragraph block section**

> Left - inner paragraph block section, Right - outer page body section

<img width="1738" alt="Screen Shot 2022-09-11 at 2 49 49 pm" src="https://user-images.githubusercontent.com/1396140/189513085-8445f730-9265-425d-a3c2-553e2cfea948.png">

**See DOM inspector - H3 is empty**

<img width="1647" alt="Screen Shot 2022-09-11 at 2 43 08 pm" src="https://user-images.githubusercontent.com/1396140/189513101-53b99e75-487a-4f61-8c25-d971f1cba20a.png">

### Potential fix

https://github.com/wagtail/wagtail/blob/519caf25e9d088adb436cff66d4a82e5a35cbeed/client/src/components/StreamField/blocks/BaseSequenceBlock.js#L151-L154

My guess is that the lines above can merge into one, the div block__type and h3 header__title can be the same content, and the div can simply be removed.

This will come with some styling fixes to do also.

### Technical details

-   Python version: 3.10
-   Django version: 4.1
-   Wagtail version: 4.0 & also checked on latest master (4.1a)
-   Browser version: Tested on Firefox 104, Safari 15 on MacOS with VoiceOver, Edge 99 on Windows 10 with NVDA

**Comment by thibaudcolas** [comment]
Thank you for filing this LB! I had stumbled upon this as well through some earlier testing, and found this had been done on purpose – this element contains the block’s "data summary" when collapsed, and is intentionally empty when expanded:

https://github.com/wagtail/wagtail/blob/d1908ee07f3959abfdd1d317ff2fa3598a27579e/client/src/components/StreamField/blocks/BaseSequenceBlock.js#L336-L350

I believe we’ll still likely need two elements (one for the block type, one for the block’s summary content), but they need to be inside the `<h3>`.

And – not 100% sure about this but I think it’d also make sense for the summary content to always be available for screen reader users.

**Comment by thibaudcolas** [comment]
This has been fixed in #9319.
